include/boost/geometry/algorithms/detail/sections/section_functions.hpp

```diff
@@ -131,7 +132,14 @@ static inline bool preceding(int dir,
                              RobustBox const& other_robust_box,
                              RobustPolicy const& robust_policy)
 {
-    typename geometry::robust_point_type<Point, RobustPolicy>::type robust_point;
+    typedef typename geometry::robust_point_type<Point, RobustPolicy>::type robust_point_type;
+    BOOST_STATIC_ASSERT((boost::is_same
+                         <
+                             typename geometry::coordinate_type<robust_point_type>::type,
+                             typename geometry::coordinate_type<RobustBox>::type
+                         >::value));
```

Comment:
This should make applying wrong types impossible

Reply:
We can use `std::is_same` now. ;) But none of the included files probably include `<type_traits>`, so let's leave it as it is until my PR is merged and I'll change this afterwards.